### PR TITLE
fix(break_streaming_task_and_rebuild): ignore reboot exception after decommission

### PR DESCRIPTION
**Pull request by amoskong**
If decommission completes, the reboot will fail for the follow err:

- init - Startup failed: std::runtime_error (This node was decommissioned and
  will not rejoin the ring unless override_decommission=true has been set,or
  all existing data is removed and the node is bootstrapped again)

It's expected, this patch changed to ignore the reboot exception after
decommission.

Signed-off-by: Amos Kong <amos@scylladb.com>
Reported-by: Shoshan Dagany <shoshan.dagany@scylladb.com>

Fixes https://github.com/scylladb/scylla-cluster-tests/issues/3015

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -2347,10 +2347,18 @@ def is_streaming_started():
...
-        self.target_node.reboot(hard=True, verify_ssh=True)
-        streaming_thread.join(60)
+        try:
+            self.target_node.reboot(hard=True, verify_ssh=True)
+            streaming_thread.join(60)
+        except Exception as err:  # pylint: disable=broad-except
+            if task == 'decommission':
+                self.log.debug(f'Failed to reboot the target node, decommission might complete')
+            else:
+                raise
```
Comment (fgelcer) [inline sdcm/nemesis.py:2357]: raise what? maybe `raise err`? or raise a better message for user to be able to catch why it failed?

**Review by fgelcer — CHANGES_REQUESTED** [review]

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -2347,10 +2347,18 @@ def is_streaming_started():
         time.sleep(random.randint(10, 600))
 
         self.log.debug('Interrupt the task by hard reboot')
-        self.target_node.reboot(hard=True, verify_ssh=True)
-        streaming_thread.join(60)
+        try:
+            self.target_node.reboot(hard=True, verify_ssh=True)
+            streaming_thread.join(60)
+        except Exception as err:  # pylint: disable=broad-except
```
Comment (fruch) [inline sdcm/nemesis.py:2353]: can we be more specific on the failure we are expecting here ?

File: sdcm/nemesis.py (hunk 1 above)
Comment (fruch) [reply sdcm/nemesis.py:2357]: open `raise` is o.k. it would raise the current exception that was caught

**Comment by fgelcer** [comment]
@amoskong , any new with it?
longevity-100gb-4h builds 26 and 27 failed with this very same issue.

@fruch , should we disable the nemesis until we get this PR sorted out?

**Comment by fruch** [comment]
@amoskong this won't work, i.e. there an Event trigger cause of the db log, not an exception...
(I've looked now on the logs of one of those happening on master)

**Comment by fruch** [comment]
@amoskong, this happening on lots of master runs, and on 4.3 too. 
we just need to add a filter to ignore the error from the log here...

**Pull request closed without merging**